[RFC] Feature bot

makidelille opened the pull request:
add Telecoms diagnostics
add "WhoAmI" Intent
externalize local strings
beautify user display in slack/messenger

Commit:
:sparkles: feat(diag): add xdsl diagnostic

Commit:
:sparkles: feat(bot): add whoAmI

Commit:
:tractor: refactor(diag): better explanation

Commit:
:sparkles: feat(diag): add telephony diagnostic

Commit:
:tractor: refactor(diag): change button url to inline urls

Commit:
:sparkles: feat(bot): externalise local strings => constants

antleblanc commented:
Sounds good for adding thoses messages inside a CHANGELOG.md file nope? @bnjjj are you agree with that?

antleblanc commented on bots/messageTypes/message/telephony_break.js:
```diff
@@ -0,0 +1,27 @@
+"use strict";
+
+// const Bluebird = require("bluebird");
```
cut?

antleblanc commented on constants/guides.js:
```diff
@@ -2,6 +2,7 @@
 
 module.exports = {
   FR: {
+    help: (guide) => `Je te conseille de consulter ce guide pour t'aider: ${guide}`,
```
`t'aider:` => `t’aider :` (space before `:`)

antleblanc commented on constants/responses.js:
```diff
@@ -2,15 +2,60 @@
+    feedbackNo: "Non",
+    feedbackThanks: "Merci pour votre avis",
+    feedbackYes: "Oui",
+    goToManager: "Acceder au manager",
```
`au manager` => `à l’espace client`

antleblanc commented on constants/responses.js:
```diff
@@ -2,15 +2,60 @@
+    hostingWrongSite: "Tu as du sélectionner le mauvais hébergement web associé à ce domaine.",
+    hostingSuspended: "Ton service hébergement web semble être suspendu, pour le réactiver il faut le renouveler via le manager",
+    ndhGetQR: "Obtenir le QR Code",
+    ndhQuestionStart: "Salut à toi 👋, tu as soif ? 🍺 Il faut la mériter si tu en veux. ",
```
Extra space at the end of the line?

makidelille replied:
intended, as in `message/ndh_question.js`, there is a concatenation with ndhQuestion :)

antleblanc replied:
Ok, thx.

antleblanc commented on constants/responses.js:
```diff
@@ -2,15 +2,60 @@
+      "Combien de langues sont disponibles sur le site ovh.com ?",
+      "Que veut dire OVH ?",
+      "Quelle est ma version ? Petit indice: je suis open source",
+      "Combien j'ai de contributeurs (sur github)?",
```
`Combien j'ai de contributeurs (sur github)?` => `Combien ai-je de contributeurs (sur GitHub) ?`

antleblanc commented on constants/responses.js:
```diff
@@ -2,15 +2,60 @@
+      "Que veut dire OVH ?",
+      "Quelle est ma version ? Petit indice: je suis open source",
+      "Combien j'ai de contributeurs (sur github)?",
+      "Combien de partenaires techniques possède OVH ? indice: https://www.ovh.com/fr/apropos/partenaires-techniques.xml",
```
`indice:` => `indice :`

antleblanc commented on constants/responses.js:
```diff
@@ -2,15 +2,60 @@
+      "Où se situe le siège social d'OVH ?",
+      "Combien de langues sont disponibles sur le site ovh.com ?",
+      "Que veut dire OVH ?",
+      "Quelle est ma version ? Petit indice: je suis open source",
```
`indice:` => `indice :`

antleblanc commented on constants/responses.js:
```diff
@@ -2,15 +2,60 @@
+    ndhWin: "Bravo tu as trouvé, voici ta récompense :)",
+    noAnswer: "Je ne peux pas encore répondre à cette question",
     noIntent: "Je n'ai pas bien compris ta demande, pour voir la liste des questions auxquelles je peux répondre consulte ce site https://www.ovh.com/manager/web/#/labs/chatbot",
+    notConnected: "Tu n'es pas connecté, mais tu peux te connecter en me demandant: 'connecte moi'",
```
`me demandant:` => `me demandant :`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,81 @@
+      errorConnRefused: "Il semblerait que ce soit un problème de pointage sur ton site, je te conseille de vérifier ta configuration DNS pour que ton nom de domaine pointe sur l'ip %s",
+      errorNotFound: "Il semblerait que ta zone dns soit mal configurée",
+      errorEaiAgain: "Il semblerait que ta zone dns soit mal configurée. Ton site n'est relié à aucun serveur d'hébergement web.",
+      goToManager: "Accéder au manager",
```
`au manager` => `à l’espace client`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,81 @@
+      accountDeleted: "Votre compte a été supprimé",
+      accountExpired: "Votre compte est expiré",
+      lineInCreation: "Votre ligne est en cours de création",
+      lineUnPaid: "Il semblerait que vous avez oublié de payer, rendez vous dans votre manager pour gérer la facturation : %s",
```
`manager` => `espace client`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,81 @@
+      portabilityProgress: "Vous avez %s lignes en cours de portabilité",
+      portabilityLineOperator: "%1s chez %2s:\n",
+      portabilityStep: "\t -%1s, Etat: %2s, Description: %3s, ETA: %4s\n",
+      portabilityExecutionDate: "Date d'éxécution prévu: %s",
```
`prévu:` => `prévu :`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,81 @@
+      portabilityLineOperator: "%1s chez %2s:\n",
+      portabilityStep: "\t -%1s, Etat: %2s, Description: %3s, ETA: %4s\n",
+      portabilityExecutionDate: "Date d'éxécution prévu: %s",
+      portabilityManager: "`Pour modifer votre portabilité, rendez vous dans votre manager: %s",
```
`manager` => `espace client`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,81 @@
+      portabilityExecutionDate: "Date d'éxécution prévu: %s",
+      portabilityManager: "`Pour modifer votre portabilité, rendez vous dans votre manager: %s",
+      noIssue: "Nous n'avons pas trouver de probleme",
+      seeManager: "Pour tout autres renseignements, rendez vous dans votre manager: %s"
```
`manager` => `espace client`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,81 @@
+      diagnosticTime: "Diagnostique réalisé à %s\n",
+      callSupport: "Appeler le service client",
+      launchDiag: "Effectuer un diagnostique avancé",
+      diagnosticModemUnplug: "\t- Votre modem n'est pas branché, vérifier la prise electrique ainsi que la prise xdsl\n",
```
`electrique` => `électrique`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,81 @@
+      launchDiag: "Effectuer un diagnostique avancé",
+      diagnosticModemUnplug: "\t- Votre modem n'est pas branché, vérifier la prise electrique ainsi que la prise xdsl\n",
+      diagnosticModemPlug: "\t- Votre modem est connecté\n",
+      diagnosticLineStatus: "\t- Etat de vos lignes:\n",
```
`lignes:` => `lignes :`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,81 @@
+      diagnosticLineStatus: "\t- Etat de vos lignes:\n",
+      diagnosticLineSync: "\t\t• %1s: %2s\n",
+      diagnosticCustomerSideProblem: "\t\t\tIl s'agit d'un probleme avec votre installation\n",
+      diagnosticOvhSideProblem: "\t\t\tIl s'agit d'un probleme de notre part\n",
```
`probleme` => `problème`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,81 @@
+      diagnosticModemPlug: "\t- Votre modem est connecté\n",
+      diagnosticLineStatus: "\t- Etat de vos lignes:\n",
+      diagnosticLineSync: "\t\t• %1s: %2s\n",
+      diagnosticCustomerSideProblem: "\t\t\tIl s'agit d'un probleme avec votre installation\n",
```
`probleme` => `problème`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,81 @@
+      diagnosticError: "\t\t\tIl y a eu une erreur lors du diagnostique\n",
+      diagnosticPing: "\t- Votre modem ne répond pas aux pings\n",
+      diagnosticNoPing: "\t- Votre modem répond aux pings\n",
+      diagnosticResultOk: "Nous n'avons pas detecter de probleme. Votre soucis provient probablement de votre installation\n",
```
`probleme` => `problème`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,81 @@
+      diagnosticPing: "\t- Votre modem ne répond pas aux pings\n",
+      diagnosticNoPing: "\t- Votre modem répond aux pings\n",
+      diagnosticResultOk: "Nous n'avons pas detecter de probleme. Votre soucis provient probablement de votre installation\n",
+      diagnosticResultNOk: "Il y a effectivement un soucis, sur votre installation\n",
```
-`,`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,81 @@
+      diagnosticResultMore: "Si votre ligne ne fonctionne toujours pas, rendez-vous sur : https://docs.ovh.com/display/public/CRXDSL/Interruption+de+service\n",
+      incident: "il y a un incident, sur la ligne : %1s, la résolution est prévue pour : %2s, Plus de Details : %3s",
+      orderStepStatus: "l'étape %1s n'est pas encore finie, elle est actuellement dans l'état : %2s. Cette étape sera finie dans %3s.\n",
+      orderNotReady: "Votre commande n'est pas encore finalisée: %s",
```
`finalisée:` => `finalisée :`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,81 @@
+      orderStepStatus: "l'étape %1s n'est pas encore finie, elle est actuellement dans l'état : %2s. Cette étape sera finie dans %3s.\n",
+      orderNotReady: "Votre commande n'est pas encore finalisée: %s",
+      lineSlamming: "Il semblerait qu'un autre opérateur ait ouvert une ligne par dessus votre ligne actuel",
+      lineUnPaid: "Il semblerait que vous avez oublié de payer, accéder au manage : https://www.ovhtelecom.fr/manager/index.html#/",
```
`au manage` (typo) => `à l’espace client`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,81 @@
+      orderNotReady: "Votre commande n'est pas encore finalisée: %s",
+      lineSlamming: "Il semblerait qu'un autre opérateur ait ouvert une ligne par dessus votre ligne actuel",
+      lineUnPaid: "Il semblerait que vous avez oublié de payer, accéder au manage : https://www.ovhtelecom.fr/manager/index.html#/",
+      resultOk: "Nous n'avons pas detecter de probleme sur votre facturation et sur la ligne.",
```
`probleme` => `problème`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,81 @@
+      lineSlamming: "Il semblerait qu'un autre opérateur ait ouvert une ligne par dessus votre ligne actuel",
+      lineUnPaid: "Il semblerait que vous avez oublié de payer, accéder au manage : https://www.ovhtelecom.fr/manager/index.html#/",
+      resultOk: "Nous n'avons pas detecter de probleme sur votre facturation et sur la ligne.",
+      resultAdvancedDiag: "Souhaitez vous faire un diagnostique plus precis de votre équipement ?\n\n Assurez vous d'avoir bien brancher votre box",
```
`Souhaitez vous` => `Souhaitez-vous`
`Assurez vous` => `Assurez-vous`
`d'avoir bien brancher` => `d'avoir bien branché`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,81 @@
+      lineUnPaid: "Il semblerait que vous avez oublié de payer, accéder au manage : https://www.ovhtelecom.fr/manager/index.html#/",
+      resultOk: "Nous n'avons pas detecter de probleme sur votre facturation et sur la ligne.",
+      resultAdvancedDiag: "Souhaitez vous faire un diagnostique plus precis de votre équipement ?\n\n Assurez vous d'avoir bien brancher votre box",
+      resultLastDiag: "Voici le résultat du dernier diagnostique réalisé. ",
```
Extra space at the end of the line?

antleblanc commented on constants/responses.js:
```diff
@@ -2,15 +2,60 @@
+    signInFirst: "Tu dois d'abord te connecter. Pour ce faire tu peux me le demander",
+    slackFallback: "Vous ne pouvez pas utiliser cette fonctionnalité avec ce navigateur",
+    telephonySelectAccount: "Selectionnez votre compte",
+    telephonyNoAccount: "Vous n'avez pas d'offre telephonie.",
```
`telephonie` => `téléphonie`

antleblanc reviewed (commented):
Quick feedback about sentences/typos and voilà! Feel free to tell me if I'm wrong.

bnjjj commented on bots/messageTypes/message/xdsl_break.js:
```diff
@@ -0,0 +1,31 @@
+    return utils
+      .getOvhClient(senderId)
+      .then((lUser) => {
+        user = lUser;
```
why do you call that `user` ? I think it could be weird, less readable

bnjjj commented on bots/messageTypes/message/xdsl_break.js:
```diff
@@ -0,0 +1,31 @@
+        feedback: false
+      }))
+      .catch((err) => {
+        Bluebird.reject(error(err.error || err.statusCode || 400, err));
```
return or inline

bnjjj commented on bots/messageTypes/postback/telephony.js:
```diff
@@ -0,0 +1,68 @@
+      let service = postback.match(new RegExp(regx))[1];
+
+      return utils.getOvhClient(senderId)
+      .then((user) => Bluebird.props({
```
same, why `user`?

bnjjj commented on bots/messageTypes/postback/telephony.js:
```diff
@@ -0,0 +1,68 @@
+          .then((arr) => arr.map((id) => {
+            let portability;
+
+            return user.requestPromised("GET", `/telephony/${service}/portability/${id}`)
```
split into function to improve readability

bnjjj commented on bots/messageTypes/postback/telephony.js:
```diff
@@ -0,0 +1,68 @@
+
+      return utils.getOvhClient(senderId)
+      .then((lUser) => {
+        user = lUser;
```
//

bnjjj commented on bots/messageTypes/postback/xdsl.js:
```diff
@@ -0,0 +1,113 @@
+            orderFollowUp: user.requestPromised("GET", `/xdsl/${xdslOffer}/orderFollowup`),
+            incident: user.requestPromised("GET", `/xdsl/${xdslOffer}/incident`)
+              .catch((err) => {
+                if (err.error === 404 || err.errorCode === 404 || err.statusCode === 404) {
```
function ? because it's duplicate

bnjjj commented on bots/messageTypes/postback/xdsl.js:
```diff
@@ -0,0 +1,113 @@
+        .map((offer) => user.requestPromised("GET", `/xdsl/${offer}`)
+          .then((xdslInfo) => new Button(BUTTON_TYPE.POSTBACK, `XDSL_SELECTED_${xdslInfo.accessName}`, xdslInfo.description))
+        )
+        .then((buttons) => ({ responses: createPostBackList(responsesCst.xdslSelect, buttons, "MORE_XDSL", parseInt(postback.match(new RegExp(regx))[1], 10), 10), feedback: false }))
```
there is a useles `,10)` I think

makidelille replied:
you're right, but eslint ;)

bnjjj commented on controllers/authorize.js:
```diff
@@ -32,7 +32,7 @@ module.exports = function () {
       return ovhClient
         .requestPromised("POST", "/auth/credential", {
-          accessRules: [{ method: "GET", path: "/*" }],
+          accessRules: [{ method: "GET", path: "/*" }, { method: "POST", path: "/*" }],
```
could you add more restriction ? Just add POST on /telephony for example ?

bnjjj commented on controllers/messenger.js:
```diff
@@ -199,7 +199,9 @@ module.exports = () => {
   function sendResponses (res, senderId, responses) {
-    return Bluebird.mapSeries(responses, (response) => sendResponse(res, senderId, response));
+    return Bluebird.mapSeries(responses, (response) =>
+      Bluebird.resolve(response)
```
Bluebird.resolve here is useless

makidelille replied:
nope, responses is not necessarily a promise :)

bnjjj commented on controllers/slack.js:
```diff
@@ -188,7 +176,9 @@ function sendQuickResponses (res, senderId, responses, slack) {
 function sendResponses (res, channel, responses, slack) {
-  return Bluebird.mapSeries(responses, (response) => sendResponse(res, channel, response, slack));
+  return Bluebird.mapSeries(responses, (response) =>
+    Bluebird.resolve(response)
```
Useless

makidelille replied:
same

bnjjj commented on diagnostics/telephony.js:
```diff
@@ -0,0 +1,68 @@
+      let portabilityString = diagCst.portabilityLineOperator.replace("%1s", porta.numbersList.join(", ")).replace("%2s", porta.operator);
+      for (let j = 0; j < porta.status.length; j++) {
+        let step = porta.status[i];
+        portabilityString += diagCst.portabilityStep.replace("%1s", step.name).replace("%2s", step.status).replace("%3s", step.description || "N/A").replace("%4s", `${step.duration.quantity} ${step.duration.unit}`);
```
Maybe there is a lib which do better job ?

Commit:
chore(travis): add travis.yml

Signed-off-by: Benjamin Coenen <benjamin.coenen@corp.ovh.com>

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+    FR: {
+      dns: "Ton site ne pointe pas sur la bonne ip (actuellement %1$s), ton domaine \"%2$s\" devrait pointer sur l'ip %3$s",
+      errorConnRefused: "Il semblerait que ce soit un problème de pointage sur ton site, je te conseille de vérifier ta configuration DNS pour que ton nom de domaine pointe sur l'ip %s",
+      errorNotFound: "Il semblerait que ta zone dns soit mal configurée",
```
`dns` => `DNS` (same for the next line.)

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      errorConnRefused: "Il semblerait que ce soit un problème de pointage sur ton site, je te conseille de vérifier ta configuration DNS pour que ton nom de domaine pointe sur l'ip %s",
+      errorNotFound: "Il semblerait que ta zone dns soit mal configurée",
+      errorEaiAgain: "Il semblerait que ta zone dns soit mal configurée. Ton site n'est relié à aucun serveur d'hébergement web.",
+      goToManager: "Accéder au espace client",
```
`au espace client` => `à l’espace client`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      goToManager: "Accéder au espace client",
+      hostingActive: "Ton site semble fonctionner correctement",
+      hostingButActive: "Par contre ton site semble fonctionner correctement",
+      hostingBloqued: "Il semblerait qu'il y ai un prolème sur ton serveur mutualisé, il est bloqué pour l'instant",
```
`prolème` => `problème`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      mailBounce: "Il semblerait que tes envois d'e-mails soient bloqués car ils ont dépassés 50% de retour en erreur",
+      mailKo: "Il semblerait que tes envois d'e-mails soient bloqués car ils ont dépassés 5% de retour en erreur",
+      mailSpam: "Il semblerait que tes envois d'e-mails soient bloqués car ils ont envoyés des spams",
+      sslRegenerate: "Tu as activé le SSL sur ton site mais ton certificat SSL ne contient pas ton domaine. Tu dois regénérer ton certificat via le espace client",
```
`le espace client` => `l’espace client`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+        " ton site ne sera plus accessible en https." +
+        " Si c'est le comportement voulu ou que tu n'utilises pas https sur ce site ne prend pas en compte cet avertissement.",
+      web404: "Il semblerait que la page à laquelle tu souhaites accéder n'existe pas, le fichier correspondant à cette page semble introuvable. " +
+        "Vérifie aussi si ton site pointe sur le bon dossier de destination dans le espace client onglet 'multisites'.",
```
`le espace client` => `l’espace client`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      web404: "Il semblerait que la page à laquelle tu souhaites accéder n'existe pas, le fichier correspondant à cette page semble introuvable. " +
+        "Vérifie aussi si ton site pointe sur le bon dossier de destination dans le espace client onglet 'multisites'.",
+      web401: "Il semblerait que ton site essaie d'accéder à un fichier ou à un espace auquel il n'a pas les droits",
+      web403: "Il semblerait que ton site possède des problèmes de droits. Vérifie si ton site pointe sur le bon dossier de destination dans le espace client onglet 'multisites'.",
```
`le espace client` => `l’espace client`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      diagnosticError: "\t  Il y a eu une erreur lors du diagnostique\n",
+      diagnosticPing: "- Votre modem ne répond pas aux pings\n",
+      diagnosticNoPing: "- Votre modem répond aux pings\n",
+      diagnosticResultOk: "Nous n'avons pas detecter de problème. Votre soucis provient probablement de votre installation\n",
```
`detecter` => `détecté`

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      diagnosticPing: "- Votre modem ne répond pas aux pings\n",
+      diagnosticNoPing: "- Votre modem répond aux pings\n",
+      diagnosticResultOk: "Nous n'avons pas detecter de problème. Votre soucis provient probablement de votre installation\n",
+      diagnosticResultNOk: "Il y a effectivement un soucis, sur votre installation\n",
```
Could you please remove extra character `,`.

antleblanc commented on constants/responses.js:
```diff
@@ -2,15 +2,59 @@
+    feedbackNo: "Non",
+    feedbackThanks: "Merci pour votre avis",
+    feedbackYes: "Oui",
+    goToManager: "Acceder à l'espace client",
```
`Acceder` => `Accéder`

antleblanc commented on constants/responses.js:
```diff
@@ -2,15 +2,59 @@
+    domainEditDns: "Tu dois modifier la zone DNS de ton domaine afin d'y ajouter le champs A avec l'ip de ton hébergement web, cette ip se trouve sur la page d'informations de ton hébergement dans l'espace client OVH.",
+    disconnected: "Tu n'es pas correctement connecté à ton compte OVH :( , il te suffit de me demander 'connecte moi' pour te reconnecter.",
+    dnsEditDns: "Tu peux modifier tes serveurs via l'espace client OVH",
+    feedbackBadUnderstanding: "Mauvaise comprehension",
```
`comprehension` => `compréhension`

antleblanc commented on constants/responses.js:
```diff
@@ -2,15 +2,59 @@
+      "Combien de langues sont disponibles sur le site ovh.com ?",
+      "Que veut dire OVH ?",
+      "Quelle est ma version ? Petit indice : je suis open source",
+      "Combien ai-je de contributeurs (sur github)?",
```
`github` => `GitHub`

antleblanc commented on constants/responses.js:
```diff
@@ -2,15 +2,59 @@
+    signIn: "Se connecter à son compte OVH",
+    signInFirst: "Tu dois d'abord te connecter. Pour ce faire tu peux me le demander",
+    slackFallback: "Vous ne pouvez pas utiliser cette fonctionnalité avec ce navigateur",
+    telephonySelectAccount: "Selectionnez votre compte",
```
`Selectionnez` => `Sélectionnez`

antleblanc commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "OvhChatbot",
-  "version": "1.1.0",
+  "version": "1.0.0",
```
@bnjjj semver issue?

makidelille replied:
intended ;)

Commit:
:sparkles: feat(slack/messenger): beautify user display

Commit:
:handshake: chore(merge)

Conflicts:
	bots/messageTypes/postback/hosting.js
	platforms/messenger/messenger_adapters.js

Commit:
fix(slack): send 200 when receving a message and not prossed

Commit:
fix(diag): format display

Commit:
:tractor: refactor(package): change v => { sprintf }, easier to read

Commit:
feat(diagnostic/hosting): add dns verify

Commit:
feat(platform): emojify !

Commit:
fix(diag/xdsl): add manager link

Commit:
:handshake: chore(merge): Merge branch 'master' of github.com:ovh-ux/ovh-chatbot into feat_diag_telecom

Conflicts:
	bots/utils.js

Commit:
fix(web): security issue with web model add ttl

antleblanc commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      orderNotReady: "Ta commande n'est pas encore finalisée : %1$s\nRends toi dans ton espace client pour plus de details :point_right: %2$s",
+      lineSlamming: "Il semblerait qu'un autre opérateur ait ouvert une ligne par dessus ta ligne actuel",
+      lineUnPaid: "Il semblerait que tu as oublié de payer, accéder à l'espace client :point_right: https://www.ovhtelecom.fr/manager/index.html#/",
+      resultOk: "Nous n'avons pas detecter de problème sur ta facturation et sur la ligne. :v:\nTu peux te rendre dans ton espace clien :point_right: %s",
```
`clien` => `client`

Commit:
fix(diag): web support

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      goToManager: "Accéder à l'espace client :point_right:",
+      hostingActive: "Ton site semble fonctionner correctement :v:",
+      hostingButActive: "Par contre ton site semble fonctionner correctement :v:",
+      hostingBloqued: "Il semblerait qu'il y ai un problème sur ton serveur mutualisé, il est bloqué pour l'instant :skull:",
```
Il semblerait qu'il y ai -> Il semble qu'il y ait

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      hostingActive: "Ton site semble fonctionner correctement :v:",
+      hostingButActive: "Par contre ton site semble fonctionner correctement :v:",
+      hostingBloqued: "Il semblerait qu'il y ai un problème sur ton serveur mutualisé, il est bloqué pour l'instant :skull:",
+      hostingMaintenance: "Il semblerait qu'il y ai un problème sur ton serveur mutualisé, il est actuellement en maintenance :wrench:",
```
same here

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      hostingBloqued: "Il semblerait qu'il y ai un problème sur ton serveur mutualisé, il est bloqué pour l'instant :skull:",
+      hostingMaintenance: "Il semblerait qu'il y ai un problème sur ton serveur mutualisé, il est actuellement en maintenance :wrench:",
+      hostingUnknown: "Ton serveur est dans un état inconnu, contacte le support pour plus d'informations",
+      mailBounce: "Il semblerait que tes envois d'e-mails soient bloqués car ils ont dépassés 50% de retour en erreur",
```
Il semble que... ils ont dépassé 50% de retours en erreur.

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      hostingMaintenance: "Il semblerait qu'il y ai un problème sur ton serveur mutualisé, il est actuellement en maintenance :wrench:",
+      hostingUnknown: "Ton serveur est dans un état inconnu, contacte le support pour plus d'informations",
+      mailBounce: "Il semblerait que tes envois d'e-mails soient bloqués car ils ont dépassés 50% de retour en erreur",
+      mailKo: "Il semblerait que tes envois d'e-mails soient bloqués car ils ont dépassés 5% de retour en erreur",
```
same as previous message

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      sslRegenerate: "Tu as activé le SSL sur ton site mais ton certificat SSL ne contient pas ton domaine. Tu dois regénérer ton certificat via l'espace client",
+      sslHttpsToHttpWarning: ":warning: Attention, ton site est compris dans le certificat SSL actuel et fonctionne actuellement en https. Cependant ta configuration indique que lors de la regénération automatique du certificat SSL," +
+        " ton site ne sera plus accessible en https." +
+        " Si c'est le comportement voulu ou que tu n'utilises pas https sur ce site ne prend pas en compte cet avertissement.",
```
ne prend -> ne prends

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      sslHttpsToHttpWarning: ":warning: Attention, ton site est compris dans le certificat SSL actuel et fonctionne actuellement en https. Cependant ta configuration indique que lors de la regénération automatique du certificat SSL," +
+        " ton site ne sera plus accessible en https." +
+        " Si c'est le comportement voulu ou que tu n'utilises pas https sur ce site ne prend pas en compte cet avertissement.",
+      web404: "Il semblerait que la page à laquelle tu souhaites accéder n'existe pas, le fichier correspondant à cette page semble introuvable. " +
```
How about : 
"Il semble que la page à laquelle tu souhaites accéder n'existe pas, le fichier correspondant à celle-ci est introuvable."

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+        " ton site ne sera plus accessible en https." +
+        " Si c'est le comportement voulu ou que tu n'utilises pas https sur ce site ne prend pas en compte cet avertissement.",
+      web404: "Il semblerait que la page à laquelle tu souhaites accéder n'existe pas, le fichier correspondant à cette page semble introuvable. " +
+        "Vérifie aussi si ton site pointe sur le bon dossier de destination dans l'espace client onglet 'multisites'.",
```
dans l'espace client onglet 'multisites' ->
"dans l'espace client : onglet 'multisites'" or "dans l'espace client (onglet 'multisites')"

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      web404: "Il semblerait que la page à laquelle tu souhaites accéder n'existe pas, le fichier correspondant à cette page semble introuvable. " +
+        "Vérifie aussi si ton site pointe sur le bon dossier de destination dans l'espace client onglet 'multisites'.",
+      web401: ":no_entry: Il semblerait que ton site essaie d'accéder à un fichier ou à un espace auquel il n'a pas les droits",
+      web403: ":no_entry: Il semblerait que ton site possède des problèmes de droits. Vérifie si ton site pointe sur le bon dossier de destination dans l'espace client onglet 'multisites'.",
```
possède -> ait

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+  },
+  telephony: {
+    FR: {
+      accountClosed: "Ton compte est cloturé :cry:",
```
clôturé

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      accountDeleted: "Ton compte a été supprimé :cry:",
+      accountExpired: "Ton compte est expiré :cry:",
+      lineInCreation: "Ta ligne est en cours de création",
+      lineUnPaid: "Il semblerait que tu as oublié de payer, rends-toi dans ton espace client pour gérer la facturation :point_right: %s",
```
How about :
Il semblerait que tu as oublié -> Il semble que tu aies

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      portabilityProgress: "Tu as %s lignes en cours de portabilité",
+      portabilityLineOperator: "%1$s chez %2$s :\n",
+      portabilityStep: "\t -%1$s, État: %2$s, description : %3$s, ETA: %4$s\n",
+      portabilityExecutionDate: "Date d'éxécution prévu : %s",
```
prévue

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      portabilityLineOperator: "%1$s chez %2$s :\n",
+      portabilityStep: "\t -%1$s, État: %2$s, description : %3$s, ETA: %4$s\n",
+      portabilityExecutionDate: "Date d'éxécution prévu : %s",
+      portabilityManager: "`Pour modifer ta portabilité, rends-toi dans ton espace client :point_right: %s",
```
modifer -> modifier

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      portabilityStep: "\t -%1$s, État: %2$s, description : %3$s, ETA: %4$s\n",
+      portabilityExecutionDate: "Date d'éxécution prévu : %s",
+      portabilityManager: "`Pour modifer ta portabilité, rends-toi dans ton espace client :point_right: %s",
+      noIssue: "Nous n'avons pas trouver de problème :v:",
```
trouvé

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      portabilityExecutionDate: "Date d'éxécution prévu : %s",
+      portabilityManager: "`Pour modifer ta portabilité, rends-toi dans ton espace client :point_right: %s",
+      noIssue: "Nous n'avons pas trouver de problème :v:",
+      seeManager: "Pour tout autres renseignements, rends-toi dans ton espace client :point_right: %s"
```
tout autre renseignement

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      doing: "en cours",
+      todo: "à faire",
+      error: "erreur",
+      diagnosticTime: "Diagnostique réalisé à %s\n",
```
Diagnostic

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      error: "erreur",
+      diagnosticTime: "Diagnostique réalisé à %s\n",
+      callSupport: "Appeler le service client :phone:",
+      launchDiag: "Effectuer un diagnostique avancé",
```
diagnostic

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      diagnosticTime: "Diagnostique réalisé à %s\n",
+      callSupport: "Appeler le service client :phone:",
+      launchDiag: "Effectuer un diagnostique avancé",
+      diagnosticModemUnplug: ":x: Ton modem n'est pas branché, vérifier la prise électrique ainsi que la prise xdsl :electric_plug:\n",
```
uppercase for XDSL I guess

makidelille replied:
`xDSL` ;)

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      launchDiag: "Effectuer un diagnostique avancé",
+      diagnosticModemUnplug: ":x: Ton modem n'est pas branché, vérifier la prise électrique ainsi que la prise xdsl :electric_plug:\n",
+      diagnosticModemPlug: ":heavy_check_mark: Ton modem est connecté\n",
+      diagnosticLineStatus: "- Etat de tes lignes :\n",
```
État

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      diagnosticLineSync: "\t• %1$s : %2$s\n",
+      diagnosticCustomerSideProblem: "\t  Il s'agit d'un problème avec ton installation\n",
+      diagnosticOvhSideProblem: "\t  Il s'agit d'un problème de notre part\n",
+      diagnosticError: "\t  Il y a eu une erreur lors du diagnostique\n",
```
diagnostic

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      diagnosticError: "\t  Il y a eu une erreur lors du diagnostique\n",
+      diagnosticPing: ":x: Ton modem ne répond pas aux pings\n",
+      diagnosticNoPing: ":heavy_check_mark: Ton modem répond aux pings\n",
+      diagnosticResultOk: ":heavy_check_mark: Nous n'avons pas détecté de problème. Ton soucis provient probablement de ton installation :thinking_face:\n",
```
souci

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      diagnosticPing: ":x: Ton modem ne répond pas aux pings\n",
+      diagnosticNoPing: ":heavy_check_mark: Ton modem répond aux pings\n",
+      diagnosticResultOk: ":heavy_check_mark: Nous n'avons pas détecté de problème. Ton soucis provient probablement de ton installation :thinking_face:\n",
+      diagnosticResultNOk: ":x: Il y a effectivement un soucis sur ton installation\n",
```
souci

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      diagnosticResultMore: "Si ta ligne ne fonctionne toujours pas, rends-toi sur :point_right: https://docs.ovh.com/display/public/CRXDSL/Interruption+de+service\n",
+      incident: ":construction: Il y a un incident sur la ligne : %1$s, la résolution est prévue pour : %2$s. Plus de détails :point_right: %3$s",
+      orderStepStatus: "l'étape %1$s n'est pas encore finie, elle est actuellement dans l'état : %2$s. Cette étape sera finie dans %3$s.\n",
+      orderNotReady: "Ta commande n'est pas encore finalisée : %1$s\nRends toi dans ton espace client pour plus de details :point_right: %2$s",
```
Rends-toi

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      incident: ":construction: Il y a un incident sur la ligne : %1$s, la résolution est prévue pour : %2$s. Plus de détails :point_right: %3$s",
+      orderStepStatus: "l'étape %1$s n'est pas encore finie, elle est actuellement dans l'état : %2$s. Cette étape sera finie dans %3$s.\n",
+      orderNotReady: "Ta commande n'est pas encore finalisée : %1$s\nRends toi dans ton espace client pour plus de details :point_right: %2$s",
+      lineSlamming: "Il semblerait qu'un autre opérateur ait ouvert une ligne par dessus ta ligne actuel",
```
ta ligne actuelle

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      orderNotReady: "Ta commande n'est pas encore finalisée : %1$s\nRends toi dans ton espace client pour plus de details :point_right: %2$s",
+      lineSlamming: "Il semblerait qu'un autre opérateur ait ouvert une ligne par dessus ta ligne actuel",
+      lineUnPaid: "Il semblerait que tu as oublié de payer, accéder à l'espace client :point_right: https://www.ovhtelecom.fr/manager/index.html#/",
+      resultOk: "Nous n'avons pas detecter de problème sur ta facturation et sur la ligne. :v:\nTu peux te rendre dans ton espace client :point_right: %s",
```
Nous n'avons pas detecter de problème sur ta facturation et sur la ligne. -> Nous n'avons détecté de problème ni sur ta facturation, ni sur ta ligne.

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      lineSlamming: "Il semblerait qu'un autre opérateur ait ouvert une ligne par dessus ta ligne actuel",
+      lineUnPaid: "Il semblerait que tu as oublié de payer, accéder à l'espace client :point_right: https://www.ovhtelecom.fr/manager/index.html#/",
+      resultOk: "Nous n'avons pas detecter de problème sur ta facturation et sur la ligne. :v:\nTu peux te rendre dans ton espace client :point_right: %s",
+      resultAdvancedDiag: "Souhaites-tu faire un diagnostique plus precis de ton équipement :mag: ?\n\n :zap: Assures-toi d'avoir bien branché ta box :zap:",
```
un diagnostic plus précis.
Assure-toi

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      resultOk: "Nous n'avons pas detecter de problème sur ta facturation et sur la ligne. :v:\nTu peux te rendre dans ton espace client :point_right: %s",
+      resultAdvancedDiag: "Souhaites-tu faire un diagnostique plus precis de ton équipement :mag: ?\n\n :zap: Assures-toi d'avoir bien branché ta box :zap:",
+      resultLastDiag: "Voici le résultat du dernier diagnostique réalisé.",
+      resultDiagRemaining: "Tu peux encore effectuer %s diagnostique(s) aujourd'hui."
```
diagnostic réalisé & diagnostic(s) aujourd'hui

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,82 @@
+      lineInCreation: "Ta ligne est en cours de création",
+      lineUnPaid: "Il semblerait que tu as oublié de payer, rends-toi dans ton espace client pour gérer la facturation :point_right: %s",
+      overOutplan: "Tu as dépassé le montant maximal en hors-forfait, rends-toi dans ton espace client pour modifier ta limite :point_right: %s",
+      portabilityProgress: "Tu as %s lignes en cours de portabilité",
```
ligne(s) ?

KamiAmiga commented on constants/responses.js:
```diff
@@ -2,15 +2,64 @@
   FR: {
-    noIntent: "Je n'ai pas bien compris ta demande, pour voir la liste des questions auxquelles je peux répondre consulte ce site https://www.ovh.com/manager/web/#/labs/chatbot",
+    connectedAs: "Tu es connecté en tant que %s :ok_hand:",
+    domainEditDns: "Tu dois modifier la zone DNS de ton domaine afin d'y ajouter le champs A avec l'ip de ton hébergement web, cette ip se trouve sur la page d'informations de ton hébergement dans l'espace client OVH.",
```
le champ
uppercase for IP maybe ?

KamiAmiga commented on constants/responses.js:
```diff
@@ -2,15 +2,64 @@
-    noIntent: "Je n'ai pas bien compris ta demande, pour voir la liste des questions auxquelles je peux répondre consulte ce site https://www.ovh.com/manager/web/#/labs/chatbot",
+    connectedAs: "Tu es connecté en tant que %s :ok_hand:",
+    domainEditDns: "Tu dois modifier la zone DNS de ton domaine afin d'y ajouter le champs A avec l'ip de ton hébergement web, cette ip se trouve sur la page d'informations de ton hébergement dans l'espace client OVH.",
+    disconnected: "Tu n'es pas correctement connecté à ton compte OVH :( , il te suffit de me demander 'connecte moi' pour te reconnecter.",
```
connecte-moi

Commit:
fix(constants): fr typo

KamiAmiga commented on constants/responses.js:
```diff
@@ -2,15 +2,64 @@
+    hostingSelectSite: "Sélectionne le site concerné (%1$s/%2$s)",
+    hostingSelectHost: "Sélectionne l'hébergement web sur lequel est installé ton site (%1$s/%2$s)",
+    hostingNoSite: ":globe_with_meridians: Tu n'as pas d'hébergement web :(",
+    hostingWrongSite: "Tu as du sélectionner le mauvais hébergement web associé à ce domaine. :thinking_face:",
```
Tu as dû

Or maybe you can say it this way :
Il semble que l'hébergement web que tu as sélectionné n'est pas celui associé à ce domaine.

KamiAmiga commented on constants/responses.js:
```diff
@@ -2,15 +2,64 @@
+    hostingSelectHost: "Sélectionne l'hébergement web sur lequel est installé ton site (%1$s/%2$s)",
+    hostingNoSite: ":globe_with_meridians: Tu n'as pas d'hébergement web :(",
+    hostingWrongSite: "Tu as du sélectionner le mauvais hébergement web associé à ce domaine. :thinking_face:",
+    hostingSuspended: "Ton service hébergement web semble être suspendu, pour le réactiver il faut le renouveler via l'espace client :zzz:",
```
Why don't you just say : "Ton hébergement web"

KamiAmiga commented on constants/responses.js:
```diff
@@ -2,15 +2,64 @@
+    telephonyNoAccount: ":telephone_receiver: Tu n'as pas d'offre télephonie. ",
+    thanks: "De rien avec plaisir :wink:",
     upsellingWeb: "Tu peux te commander un hébergement web en te rendant sur https://www.ovh.com/fr/hebergement-web/",
+    upsellingXDSL: "Tu peux te commander une ligne xDSL en te rendant sur https://www.ovhtelecom.fr/offre-internet/",
```
Tu peux commander

KamiAmiga commented on constants/responses.js:
```diff
@@ -2,15 +2,64 @@
   • Comment je fais pour changer mes serveurs dns de ma zone exemple.ovh ?
   • Comment je peux faire pointer mon domaine exemple.ovh sur mon hébergement web ?`,
-    welcome_logged: "Tu es bien connecté sous le nic %s :)"
+    welcomeWeb: "Bienvenue %s, en quoi puis-je être utile ?",
```
How about :"en quoi puis-je t'être utile"

Commit:
refactor(diag/xdsl): better styling

KamiAmiga commented on constants/responses.js:
```diff
@@ -2,15 +2,64 @@
+    welcomeWeb: "Bienvenue %s, en quoi puis-je être utile ?",
+    xdslSelect: "Sélectionne ton offre xDSL (%1$s/%2$s)",
+    xdslNone: ":satellite_antenna: Il semblerait que tu n'as pas d'offre xDSL.",
+    xdslDiagInProgress: "Diagnostique en cours... Patiente quelques instants, le diagnostique peut prendre queleques minutes, merci :)",
```
diagnostic

Commit:
fix(web): add response to welcome and connection intent

Commit:
refactor(constants): typos

Commit:
Merge branch 'master' of github.com:ovh-ux/ovh-chatbot into feat_diag_telecom

Commit:
:ambulance: fix(bot): fix empty title when no description in buttons

Commit:
chore(CHANGELOG): update changelog

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,94 @@
+      sslRegenerate: "Tu as activé le SSL sur ton site mais ton certificat SSL ne contient pas ton domaine. Tu dois regénérer ton certificat via l'espace client",
+      sslHttpsToHttpWarning: ":warning: Attention, ton site est compris dans le certificat SSL actuel et fonctionne actuellement en https. Cependant ta configuration indique que lors de la regénération automatique du certificat SSL," +
+        " ton site ne sera plus accessible en https." +
+        " Si c'est le comportement voulu ou que tu n'utilises pas https sur ce site, ne prend pas en compte cet avertissement.",
```
ne prends pas

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,94 @@
+      orderTreatment: "Traitement de la commande",
+      sendModem: "Envoi du modem",
+      sendOrderToProvider: "Transmission de la commande auprès du fournisseur d'accès",
+      setupCustomerPremisesEquipment: "Envoie de l'équipement",
```
Envoi

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,94 @@
+      lineSlamming: "Il semblerait qu'un autre opérateur ait ouvert une ligne par dessus ta ligne actuelle",
+      lineUnPaid: "Il semblerait que tu as oublié de payer, accéder à l'espace client :point_right: https://www.ovhtelecom.fr/manager/index.html#/",
+      resultOk: "Nous n'avons détecté de problème ni sur ta facturation, ni sur ta ligne. :v:\nTu peux te rendre dans ton espace client :point_right: %s",
+      resultAdvancedDiag: "Souhaites-tu faire un diagnostic plus precis de ton équipement :mag: ?\n\n :zap: Assure-toi d'avoir bien branché ta box :zap:",
```
précis

KamiAmiga commented on constants/responses.js:
```diff
@@ -2,15 +2,64 @@
+    disconnected: "Tu n'es pas correctement connecté à ton compte OVH :( , il te suffit de me demander 'connecte-moi' pour te reconnecter.",
+    dnsEditDns: "Tu peux modifier tes serveurs via l'espace client OVH",
+    feedbackBadUnderstanding: "Mauvaise compréhension",
+    feedbackHelp: "Est-ce que cette réponse t'as aidé ?",
```
t'a

Commit:
:ambulance: fix(mongo/web): fix security issue: missing ttl on auth

Commit:
:handshake: merge: merge branch feat_improv_dns

Commit:
chore(messenger): hide messenger send apis

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,95 @@
+      hostingUnknown: "Ton serveur est dans un état inconnu, contacte le support pour plus d'informations",
+      mailBounce: "Il semblerait que tes envois d'e-mails soient bloqués car ils ont dépassé 50% de retour en erreur",
+      mailKo: "Il semblerait que tes envois d'e-mails soient bloqués car ils ont dépassé 5% de retour en erreur",
+      mailSpam: ":no_entry_sign: Il semblerait que tes envois d'e-mails soient bloqués car ils ont envoyés des spams",
```
ont envoyé

KamiAmiga commented on constants/diagnostics.js:
```diff
@@ -0,0 +1,95 @@
+      portabilityProgress: "Tu as %s ligne(s) en cours de portabilité",
+      portabilityLineOperator: "%1$s chez %2$s :\n",
+      portabilityStep: "\t -%1$s, État: %2$s, description : %3$s, ETA: %4$s\n",
+      portabilityExecutionDate: "Date d'éxécution prévue : %s",
```
exécution

KamiAmiga commented on constants/responses.js:
```diff
@@ -2,15 +2,64 @@
-  • Comment je peux faire pointer mon domaine exemple.ovh sur mon hébergement web ?`,
-    welcome_logged: "Tu es bien connecté sous le nic %s :)"
+  • Comment je peux faire pointer mon domaine exemple.ovh sur mon hébergement web ?
+  • Peut tu faire un diagnostic de ma ligne ADSL ?
```
Peux-tu

KamiAmiga commented on constants/responses.js:
```diff
@@ -2,15 +2,64 @@
+    welcomeWeb: "Bienvenue %s, en quoi puis-je t'être utile ?",
+    xdslSelect: "Sélectionne ton offre xDSL (%1$s/%2$s)",
+    xdslNone: ":satellite: Il semblerait que tu n'as pas d'offre xDSL.",
+    xdslDiagInProgress: "Diagnostic en cours... Patiente quelques instants, le diagnostique peut prendre queleques minutes, merci :)",
```
diagnostiC ;p

KamiAmiga commented on constants/responses.js:
```diff
@@ -2,15 +2,64 @@
+    xdslSelect: "Sélectionne ton offre xDSL (%1$s/%2$s)",
+    xdslNone: ":satellite: Il semblerait que tu n'as pas d'offre xDSL.",
+    xdslDiagInProgress: "Diagnostic en cours... Patiente quelques instants, le diagnostique peut prendre queleques minutes, merci :)",
+    xdslQuotaReached: "Ton quota de diagnostic a été atteint"
```
diagnostics ?

Commit:
chore(constants): french typos

antleblanc commented on platforms/messenger/messenger.js:
```diff
@@ -135,111 +136,111 @@ function receivedAccountLink (event) {
-
-  callSendAPI(messageData);
-}
+// function sendImageMessage (recipientId) {
```
to delete?

makidelille replied:
IMHO, it will be useful for the next one, when more functions of the messenger API will be used. They are not in use for the moment, but can be useful in the future. So, I left them 😉👊

antleblanc commented on platforms/generics.js:
```diff
@@ -28,6 +36,11 @@ class ButtonsMessage {
 
+    // FB allow 3 buttons max, Reference: https://developers.facebook.com/docs/messenger-platform/send-api-reference/button-template
+    if (buttons.length > 3) {
+      throw new Error("Buttons cant be longer than 3");
```
`cant` => `can't`|| `cannot`

antleblanc commented on platforms/generics.js:
```diff
@@ -50,7 +63,12 @@ class ButtonsListMessage {
+
+
+  if (limit > MAX_LIMIT) {
+    throw new Error("Limit cant be greater than %d", MAX_LIMIT);
```
`cant` => `can't` || `cannot`
